CollAction/Controllers/ProjectsController.cs

```diff
@@ -274,14 +276,33 @@ public async Task<IActionResult> Commit(int id, CommitProjectViewModel commitPro
                     "</span>";
                 string subject = $"Thank you for participating in the \"{commitProjectViewModel.ProjectName}\" project on CollAction";
                 _emailSender.SendEmail(user.Email, subject, confirmationEmail);
-                return View("ThankYouCommit", commitProjectViewModel);
+                return LocalRedirect($"~/Projects/{Uri.EscapeDataString(commitProjectViewModel.ProjectName.NormalizeUriPart())}/{commitProjectViewModel.ProjectId}/thankyou");
             }
             else
             {
                 return View("Error");
             }
         }
 
+        [Authorize]
+        [HttpGet]
+        public IActionResult ThankYouCommit(int id, string name)
+        {
+            if (name != null)
+            {
+                CommitProjectViewModel model = new CommitProjectViewModel()
+                {
+                    ProjectId = id,
+                    ProjectName = name
```

Comment:
It's showing the "normalized/encoded" to users here, probably best if you retrieve it from the database.

Reply:
Well noticed, fixed in c519dac.